cross/ccrx.txt

```diff
@@ -6,15 +6,53 @@ c = 'ccrx'
 cpp = 'ccrx'
 ar = 'rlink'
 strip = 'rlink'
-
-[properties]
-# The '--cpu' option with the appropriate target type should be mentioned
-# to cross compile c/c++ code with ccrx,.
-c_args      = ['--cpu=rx600']
-cpp_args    = ['--cpu=rx600']
+pkgconfig = 'pkg-config'
 
 [host_machine]
-system = 'bare metal'      # Update with your system name - bare metal/OS.
+system = 'bare metal'
 cpu_family = 'rx'
 cpu = 'rx600'
 endian = 'little'
+
+[properties]
+# The '--cpu' option with the appropriate target type should be mentioned
+# to cross compile c/c++ code with ccrx,.
+needs_exe_wrapper = true
+c_args = [
+  '-cpu=rx600',
+  '-goptimize',
+  '-nostuff',
+  '-define=_REENTRANT,_FEVAL=0,_HAS_C9X_FAST_FMA=0,RX63_BOARD=CUSTOM631,__DBL8,DEVICE_PACKAGE_100_PIN,GHW_SINGLE_CHIP'
+  ]
+c_link_args = [
+  '-show=ALL',
+  '-list=device.map',
+  '-form=absolute',
+  '-nologo',
+  '-optimize',
```

Comment:
If this does the same as `-O2` does on GCC, it should not be listed in the cross file. All optimization and debug flags should come from the compiler class based on build settings, not from cross files.

Reply:
good point changed now to less info